### PR TITLE
Migrate Python scripts to Python 3

### PR DESCRIPTION
**Pull request by deajan**
Hello,

Since Python 2.x EOL is pretty close, I updated the python 2 only scripts.
List of changes:

- Python 2 to 3 migration; keeping Python 2.7 compatibility
- Changed subprocess caller to a way more robust version
- Separated procedural code from functions
- A lot of PEP-8 cleanup
- Refactored config & db code into a library
- Removed default db argument from some functions in order to prevent corner cases
- Re-tested all code
- Merge threading code into a library

This is an initial PR in order to make LibreNMS fully Python 3 compatible.
There are a lot of things left to do in order to tidy up these codes among:

- Harden the code (undeclared variable handling, better exceptions...)
- Remove that DB class which does lack abstraction layer

Willing to take a bit of time of needed.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by deajan** [comment]
Of course, codeclimate had some fixes but created new issues since code needs to be refactored.

**Comment by murrant** [comment]
Thanks for working on this @deajan   I have not reviewed it yet, but one thing to keep in mind is we still support CentOS 7 and probably have some installs on CentOS 6.  If backwards compatibility is not a part of this, it will require a large effort to notify users that polling will be affected.

**Comment by deajan** [comment]
Hmm... Backward compatibility should not be so hard to obtain here.
I'll have this tested on CentOS 7/Python 2.7 too then.

File: librenms_library.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
+#! /usr/bin/env python
```
Comment (murrant) [inline librenms_library.py]: Can you place this in a subdirectory?  There are already some scripts in the LibreNMS directory...

Comment (deajan) [reply librenms_library.py]: Sure, I was actually working on if I could merge with the existing LibreNMS/__init__.py code (at least for the database code which is pretty much the same), but there are way too much other imports in that file already.
Once I've sorted out what I can merge, I'll move the file :)

Comment (murrant) [reply librenms_library.py]: Great.

**Comment by SourceDoctor** [comment]
@deajan 
something new here?

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/dropping-support-for-php-7-1-and-python-2/11851/4

**Comment by deajan** [comment]
Sorry, I've let this slip away while improving my python skills.
I'll see if I can catch up and finish this PR soon.

**Comment by SourceDoctor** [comment]
@deajan 
on problems feel free to ask on https://community.librenms.org/
or on chat system https://discord.com/invite/V8s5JZJ 
(there exists an extra development room for questions)

**Comment by murrant** [comment]
@deajan I rebased this so github-apply will work.

- Removed some python2 support stuff, it won't be needed now.
- Found some duplciated code in services-wrapper.py
- Moved the library into the LibreNMS package

**Comment by deajan** [comment]
@murrant Any reason you deleted poller-service.py ?

**Comment by deajan** [comment]
I'll try to finish my PR this week, there are still two outstanding issues:
- lacks usage of DB class instead of that hacky query behavior
- code hardening (variables may be used before declaration

I'll update my test librenms instance this afternoon in order to play with that PR.

**Comment by murrant** [comment]
@deajan yes, it is broken and deprecated.  I don't want it ported to python3 ;)

**Comment by murrant** [comment]
@deajan I'm going to roll back your changes.  I know there is a lot of code duplication, but we need to get this merged soon and your last changes broke things.

Please start another PR with further refinements and probably start using some object oriented approach to prevent passing so many variables around.

Come contact me on Discord to discuss if you like :)

**Comment by murrant** [comment]
@deajan by the way you can restore your commits to a new branch like this:
`git checkout df97af998692a4cf0c999253a8a089c86a79cfdf -b python3-love-continued`

**Comment by deajan** [comment]
@murrant I tried to refactor a minimum to only avoid code duplication and remove some obvious caveats.

The way I see this, all three wrapper scripts should be merged into a single one, that indeed uses a class for distributed polling handling.
The downside would be that it would change the scripting bahavior, unless we add dummy scripts that would just call the main script with the right arguments.
What do you think about that idea ?

Another idea would be to replace threading with multiprocessing, so we can leverage all cpu cores power, but this would make inter process communication more diffucult.

**Comment by murrant** [comment]
I'm down for whatever you propose, but I just needed a python3 version in master before the next release ;)

**Comment by murrant** [comment]
Isn't multiprocessing the older way of doing it?  starting a process for each worker instead of a lightweight thread?

**Comment by SourceDoctor** [comment]
@murrant multiprocessing and threading are existing parallel and having both advantages and disadvantages. Threading work "parallel" but work stays on the same core. Multiprocessing opens a seperate "process" so alle CPU Cores can be real used.
Personally i use multiprocessing in most cases only if workers have to do cpu intensive stuff to do, otherwise i us threading, mostly ... for short :D

**Comment by murrant** [comment]
Honestly, there is no need to have the wrapper run on different cores is there?  All the work is done in a php process which can run on any core.

**Comment by crazy-max** [comment]
Looks like [some scripts](https://github.com/librenms/librenms/search?l=python&p=1) have been missed? `snmp-scan.py` for example?

**Comment by murrant** [comment]
@crazy-max snmp-scan.py already supports python3 ;)

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-64-release-changelog-may-2020/12290/1

**Pull request merged**